### DISCUSSION TITLE
How to implement socket communication on the server-side client nicegui

### DISCUSSION THREAD

**Discussion by liujiayi6v** [zauberzeug/nicegui#2248] [Q&A]
How to implement socket communication on the server-side client nice gui

**Comment by rodja** [comment]
Depends on what you want to do. The web socket communication between browser and Python is already handled by NiceGUI internally. If you want to communicate with other services via a plain tcp/udp socket you should be able to follow the instructions from the lib you are using. Just make sure it's doing async/await to not block the main thread.

**Comment by stepheweffie** [comment]
[https://nicegui.io/documentation/page#wait_for_client_connection](url)
Read the Docs
